### DISCUSSION TITLE
bulk edit

### DISCUSSION THREAD

**Discussion by geekpulp** [community/community#5465] [Projects and Issues] [answered]
Hi team, it's possible I just missed this in the UI of the new Beta so correct me if I'm wrong. It would be super helpful to be able to bulk edit items within a project. Say I want to select all items with 1 status and move them to another, or select all of a particular type and delete them.

Is this a thing and I've missed it?

**Comment by jackmcdade** [comment]
Yeah, looking for this too. Coming from ZenHub, i had to manually update hundreds of issues and it took a really. long. time.

**Reply by dralley** [reply]
Likewise - this is something we really miss from Redmine.

**Comment by jdhaines** [comment]
This.  I'd even settle for little checkboxes on one side that can then let you do a range of actions.

**Reply by tamalexmak** [reply]
Second this. Another use case is changing target date for a milestone. If I use a custom field with a date for issues, I'd like to bulk-select and update it.

**Comment by mitchelsellers** [comment]
Looking at converting to this process for everything I would agree that this is a huge requirement.  Ideally even if it is just that we can bulk-edit the custom fields that exist on the board it would be a good stepping stone.

**Comment by alexcnichols** [comment]
👋🏼 Improvements for bulk updates, and more, are coming soon! Check out this roadmap item for more and subscribe to be notified if/when things change. https://github.com/github/roadmap/issues/283 :octocat:

Bulk updating can have several components to it, so keep the feedback coming!

**Reply by MisinformedDNA** [reply]
I don't see any roadmap items for bulk updates. Also, I only see one feature assigned to this quarter.

**Reply by alexcnichols** [reply]
Yeah, looks like this one didn't quite make it into the improvements for December. Adding @azenMatt in case he has any updates he'd like to share about if/when this is coming.

**Reply by azenMatt** [reply]
Thanks Alex - this is still on our backlog as an improvement we want to make but I don't have a timeline to share just yet

**Reply by gf3** [reply]
@alexcnichols @azenMatt thank you for the hard work & information. is there another issue we can follow to track its progress?

**Reply by bewuethr** [reply]
https://github.com/github/roadmap/issues/650 is now specifically about this, it seems.

**Comment by chrisshaw** [comment]
I have the same need as [these 5 people](https://github.com/github/feedback/discussions/8960#discussioncomment-1916190) from the [very similar discussion on multiple select actions](https://github.com/github/feedback/discussions/8960): bulk drag and drop to shift priority for a "group" of issues.

**Comment by nkolev92** [comment]
I'd like to be able to change other fields as well. 

- bulk move issues to the new sprint (not just the next one). 
- Assign labels to a group of issues
- Assign milestone to a group of issues.

**Reply by beardwin** [reply]
This is _exactly_ my need. I tend to use the table view and just rapid-add several tasks. Then i have to go through each one individually and 
- add to the appropriate repo
- add the appropriate labels
- assign to the appropriate milestones
- etc.

It takes _way_ longer than it should. I'd ideally like a checkbox or something, and set these fields across _all_ issues.

**Comment by jhylee** [comment]
100%! It'd be great if we could select an grouping of items and:
- Adjust assignees
- Adjust status
- Change sprints

Unrelated, but would also be helpful if we could "close out" sprints so it doesn't show up in the default Planning view. I presume once we get to Sprint 15-16, it would be annoying to have to scroll down and collapse all of the tickets related to Sprint 1-15. (If this already exists, please let me know!)

**Comment by d4vidi** [comment]
FWIW, I tried introducing this myself using the fancy [GraphQL](https://docs.github.com/en/issues/trying-out-the-new-projects-experience/using-the-api-to-manage-projects) and it was pretty difficult to onboard. Would love to see this supported inherently soon 🙏🏻

**Comment by marcotas** [comment]
IMHO bulk actions are a very basic feature regarding management and I honestly thought it would exist after beta testing but it didn't happen. I'm happy for the commitment of the team to provide more features (really important ones as well) that are coming from the roadmap, but honestly, how can we "migrate" from other management software without a bulk actions feature? Github Team please, this is a MUST HAVE feature!

**Comment by airtonix** [comment]
since disucssions that duplicate other discussions are not linked (like the super useful issue tickets do), i'm repeating my closed discussion that apparently duplicates this post: 

> Frequently I want to make the same kind of changes to multiple project items(they aren't issue 
>  tickets yet):
> 
> - assign to user
> - convert to issue
> - yadda yadda yadda.
> 
> Let us make selections composed of multiple items, then provide some kind of UI that lets us 
> apply bulk changes:
> 
> - a sidebar composed of all the projects fields as facets, dragging the multiple selection of items 
> onto one of these applies the facet
> - a button that launches a modal with all the fields as their respective ui control, changing 
> one or more values here applies these changes to all selected items.

**Comment by ascourtas** [comment]
Also here to request the addition of a bulk edit feature ❤️

**Comment by kattekrab** [comment]
yes please!

**Comment by elisealix22** [comment]
I would also enjoy this feature! 😁 Would pair nicely with the shift + space keyboard shortcut!

**Comment by mbernson** [comment]
+1 for this feature request. 
As I can do in Trello, I would like to be able to move all the tickets in a certain column to a different column at once.

**Comment by kumar-slanglabs** [comment]
This is 2023 and still no bulk edit :-( Mommy... I want my bulk edit. Pretty please :-)

**Comment by tony-riotracers** [comment]
please, i'm begging, we need bulk edit.

**Comment by aa21** [comment]
@alexcnichols Hi Alex .. any updates on this?

**Comment by GregJohnStewart** [comment]
Additional bump! Please

**Comment by bewuethr** [comment]
This seems to have been [just released](https://github.blog/changelog/2023-04-06-github-issues-projects-april-6th-update/#t-rex-bulk-editing-in-tables), though I can't get it to work in my projects yet. Phased rollout maybe?

**Reply by BenjaminYde** [reply]
I have the same problem!

**Reply by ivibumblebee** [reply]
Same here

**Reply by j0siepy** [reply]
This should be working now! Let me know if it isn't.

**Comment by iamankitkathayat** [comment]
One of these answers really helped me a lot. Thanks peeps!!! :)

**Comment by j0siepy** [comment]
We've shipped our [first iteration of bulk edits](https://github.blog/changelog/2023-04-06-github-issues-projects-april-6th-update/#t-rex-bulk-editing-in-tables) in the table view. 

Please let me know if you have any feedback

**Reply by remyleone** [reply]
Does not work for me. I've tried to select several items to archive them all in one go and it didn't work.
As for the edit, I've tried to select a set of cells and it selected all the others within a view that has a grouped sorting applied.

**Reply by BenjaminYde** [reply]
Shift + RMB does only work from bottom issues to top issues, not from top to bottom (direction in which you select issues)
CTRL + Click to select multiple items works fine

When you select a cell, there is a blue marker visible, this is for copying the cells to other cells,
This works only from bottom to top, not top to bottom

Multi select cellt when keeping RMB presses is also broken when going from top to bottom

CTRL + C >> CTRL + V works fine 

When you select a cell of the most bottom row, the blue marker is only partly visible, this is because the row where you can insert new issues is rendered on top of the marker

![image](https://user-images.githubusercontent.com/44135938/231497462-278555ef-2f3f-47f2-b00a-a27972bcc5f7.png)

**Reply by j0siepy** [reply]
@remyleone in order to bulk archive follow the steps in this video. 
https://user-images.githubusercontent.com/98360703/231507474-ac094a89-fc4e-4d71-b800-a5234c857635.mov

@BenjaminYde Thank you so much for this feedback - we'll look into fixing the UX.

**Reply by remyleone** [reply]
@j0siepy Thanks it works <3

**Comment by j0siepy** [comment]
I'm really sorry folks - I had to pull back the feature because we discovered a bug. We hope to have this bug resolved in the next few days. Will provide an update as soon as I have more information. Thank you for your patience!

**Comment by iamankitkathayat** [comment]
Looking at converting to this process for everything I would agree that this is a huge requirement. Ideally even if it is just that we can bulk-edit the custom fields that exist on the board it would be a good stepping stone.

**Reply by ankitBackupKathayat** [reply]
thanks

**Comment by j0siepy** [comment, marked as answer]
The fix has been shipped and folks should be able to update multiple cells in a column at once

**Reply by aaron-schneider** [reply]
Very exciting to see this! Is there a rollout period for this? I'm not seeing it work for me to change issue status. It seems to always unselects everything when I click the status dropdown. Also, is there a way to bulk comment on those selected issues? I'd like to document the reason I'm updating the status. (I'm on Chrome in Chrome OS. I can select multiple, but not make buk changes. I've tried with the list sorted by Status and not with anything.)

**Reply by andybergon** [reply]
I guess this is for dot com, plans for GHE?

**Reply by sergeymorkovkin** [reply]
Any updates on this?

**Reply by JDMathew** [reply]
This feature would be a huge time save, any updates @j0siepy?

**Comment by BenjaminYde** [comment]
### Feature request
Would be nice to bulk close issues

**Comment by BenjaminYde** [comment]
### Feature request
Would be nice to bulk convert draft issues to issues (= assign to a repo)

**Reply by ssyberg** [reply]
> Would be nice to bulk convert draft issues to issues (= assign to a repo)

Was just looking for this, confirming this is currently impossible?

**Comment by Teammai101** [comment]
Was this ever implemented?

**Reply by michelle-bauer** [reply]
https://github.blog/changelog/2023-05-25-github-issues-projects-may-25th-update/#%F0%9F%92%AA-%E2%8C%A8%EF%B8%8F-bulk-updates-and-keyboard-navigation-on-boards

**Comment by pennypriddy** [comment]
Is there any future where we can get this as checkboxes instead of highlighted rows (which are easier to change accidentally)?

**Comment by denis-getopensocial** [comment]
Since recentlty the behaviour of the checkbox usage in the list of PR have changed.
I used to be able to bulk update PRs to assign them to someone, or update milestone.
Now when I use select the checkboxes of some PRs and click on a new milestone, it just update the search bar content.
This is very annoying that there are no bulk operation anywhere anymore, or did I miss something?

**Reply by smitmartijn** [reply]
I don't think you're missing anything - I can't find it neither,  nor can I find a change log that mentions they removed it...So here's hoping it's a mistake and it'll be added back?

**Comment by fathiraz** [comment]
hello guys, i've developed a browser extension for this challenge i'm also facing, i hope it will help you guys. the tool is inspired by a famous extension for gitHub called refined gitHub

this is the tool: https://github.com/fathiraz/refined-github-projects

you can do: bulk update field, bulk moving, even something like sprint
i hope you like it, guys!

**Reply by v-coder247** [reply]
Hi @fathiraz, thanks for sharing with the community! 


Unfortunately, we currently do not allow self-promotion, advertising, or solicitation in Community Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.
